Libraries/TranscriptTestRunner/TestRunner.cs

```diff
@@ -285,8 +291,17 @@ private async Task ExecuteTestScriptAsync(string callerName, CancellationToken c
             _logger.LogInformation($"\n------ Starting test {callerName} ----------");
 
             using var reader = new StreamReader(_testScriptPath);
+            var plainTestScript = await reader.ReadToEndAsync().ConfigureAwait(false);
+
+            if (_scriptParams?.Count > 0)
```

Comment:
Consider replacing by a more efficient alternative (we don't need to know the exact count, we just need to know if it has elements so we can iterate over it)
```suggestion
            if (_scriptParams != null && _scriptParams.Any())
```